frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/SelectionManager.java

```diff
@@ -177,14 +177,16 @@ private SelectionResult selectNodesByJobPacking(ResourceDescriptor requestResour
       if (gpuAttribute == 0) {
         gpuAttribute = selectCandidateGpuAttribute(node, requestResource.getGpuNumber());
       }
+      LOGGER.logDebug("selectNodes: Select: " + node.getHost() + " node avaliableGpuAttribute:" +
+          CommonExts.toStringWithBits(node.getAvailableResource().getGpuAttribute()) + " gpuAttribute:" + CommonExts.toStringWithBits(gpuAttribute));
       result.addSelection(node.getHost(), gpuAttribute, node.getAvailableResource().getPortRanges());
     }
     return result;
   }
 
   public synchronized SelectionResult selectSingleNode(String taskRoleName) throws NotAvailableException {
     SelectionResult results = select(taskRoleName);
-    if (results.getNodeHosts().size() > 1) {
+    if (results.getNodeHosts().size() > 0) {
       // Random pick a host from the result set to avoid conflicted requests for concurrent container requests from different jobs
```

Comment:
So this is a previous bug or by design?

Reply:
bug

---
In reply to: [200566168](https://github.com/Microsoft/pai/pull/822#discussion_r200566168) [](ancestors = 200566168)